src/tree.rs

```diff
@@ -88,6 +88,14 @@ impl MerklePath {
         }
     }
 
+    /// Instantiates a new Merkle path given a leaf position and authentication path.
+    pub(crate) fn new(position: u32, auth_path: [pallas::Base; MERKLE_DEPTH_ORCHARD]) -> Self {
```

Comment:
It occurs to me that it'd probably be good to change https://github.com/zcash/incrementalmerkletree/blob/master/src/lib.rs#L220 to have a const generic parameter for the depth, and then this type could be moved there and used as the result of finding the authentication path.